server/services/gameLogic.js

```diff
@@ -171,15 +153,32 @@ exports.voteCard = (username, cardId) => {
             cardId: cardId
         };
         votes.push(vote);
-        if (allPlayersVoted()) nextRound();
+        if (allPlayersVoted()) {
+            calcScores();
+            nextRound();
+        }
+    }
+}
+
+/* Calculate the scores for this round */
+const calcScores = () => {
+    const correctCard = playedCards.find(card => card.userId === currentPlayer.userId);
+    const correctVotes = votes.filter(vote => vote.cardId === correctCard.cardId);
+    if ((correctVotes.length % votes.length) === 0) {
+        players.forEach(player => {if (player !== currentPlayer) player.score += 2});
+    } else {
+        currentPlayer.score += 3;
+        correctVotes.forEach(vote => players.find(player => player.username === vote.username).score += 3);
+        votes.filter(vote => vote.cardId !== correctCard.cardId).map(vote => playedCards.find(card => vote.cardId === card.cardId)).forEach(card => players.find(player => player.username === card.username).score += 1);
```

Comment:
Why have a `map` and a `forEach`? Would it make more sense to drop the `map` and more the logic into one function?